src/sp/it/pl/util/file/Util.kt

```diff
@@ -71,37 +67,29 @@ fun File.isAnyChildOf(parent: File) = parent.isAnyParentOf(this)
  *
  * @return child files of the directory or empty if parameter null, not a directory or I/O error occurs
  */
-@Suppress("DEPRECATION")
```

Comment:
Hmm, but this will look very weird for anybody looking at the code. Why deprecate a JVM method? You are already aware of the behaviour, so simply use your utility functions where applicable.

Reply:
Everyone must be aware of the danger. I think we do not need to debte the usefulness of deprecation.

Even if I'm aware of the dangers (returning nul), this is not covered by Kotlin typesystem because the return type is platform type (i.e. File[]!) and does not force developer to handle null even in Kotlin. I've already been burned by this and decided the method should be wrapped around and never be called directly in the project.

Another such method is Optional.get(). There is never good justification to call get() on an Optional. The method was a major design mistake. 

It is only natural to denote and isolate these methods and IntelliJ provides a way - deprecating them with an 'external annotation' (https://www.jetbrains.com/help/idea/external-annotations.html).

I'd like to have this automatically set up with the project, but didnt have time to investigate if and how it is possible.

Reply:
I think you simply need to track the file `annotations.xml` that should be somewhere in your `.idea` folder in git, then everybody will have them. So, if it's right at the root, it would be `git add -f .idea/annotations.xml` and then commit the file directly to master.

Reply:
You are correct. But everyone would still have to manually add the file to the list of external annotation files.

Reply:
do they? Files in the `.idea` directory should be picked up by IDEA automatically, it works the same for codeStyles, you don't have to manually add them if they're in the right place, since IDEA automatically reads that file at startup anyways.

Reply:
I was under impression you have to add external annotation file manually. At least I had to. But its worth a try.